### PR TITLE
Added AddDomainIntegrator with marker for (Par)ComplexLinearForm and (Par)SesquilinearForm

### PR DESCRIPTION
**Pull request by mdeuse**
Added AddDomainIntegrator with marker for (Par)ComplexLinearForm and (Par)SesquilinearForm

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2570](https://github.com/mfem/mfem/pull/2570) | @mdeuse | @pazner | @mlstowell + @psocratis | 11/01/21 | 11/10/21 | 11/30/21 | |

### CODE REVIEW

**Comment by pazner** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by psocratis — APPROVED** [review]

File: fem/complex_fem.hpp [hunk 1]
```diff
@@ -464,6 +474,11 @@ class ParComplexLinearForm : public Vector
    void AddDomainIntegrator(LinearFormIntegrator *lfi_real,
                             LinearFormIntegrator *lfi_imag);
 
+   /// Adds new Domain Integrator.
```
Comment (mlstowell) [inline fem/complex_fem.hpp:477]: Minor point but this comment should also mention the restriction to certain attributes

**Review by mlstowell — APPROVED** [review]
Thanks @mdeuse for this addition! Once that small addition to the comment is made I think this is ready to go.

**Comment by mdeuse** [comment]
Hi @mlstowell , I have updated the comment, I think that the PR should be ready to be merged now.

**Comment by pazner** [comment]
Merged in `next` for testing...

**Comment by pazner** [comment]
@mdeuse, can you merge `master` into your branch? Thanks

**Comment by mdeuse** [comment]
Hi @pazner , I just did the merge.

**Pull request merged**